tests/regression/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+        .expect("Unable to set a host verify callback.");
+    builder
+        .load_pem(keypair.cert(), keypair.key())
+        .expect("Unable to load cert/pem");
+    builder.trust_pem(keypair.cert()).expect("load cert pem");
+    builder.build()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crabgrind as cg;
+    use s2n_tls::testing::TestPair;
+    use std::{
+        env,
+        fs::{create_dir, File},
+        io::{self, BufRead, Write},
+        path::Path,
+        process::Command,
+    };
+
+    const COST: u64 = 1_000_000; //configurable threshold for regression
+
+    struct InstrumentationControl;
+
+    impl InstrumentationControl {
+        fn stop_instrumentation(&self) {
+            cg::cachegrind::stop_instrumentation();
+        }
+
+        fn start_instrumentation(&self) {
+            cg::cachegrind::start_instrumentation();
+        }
+    }
+    // environment variable to determine whether to run under valgrind or solely test functionality
+    fn is_running_under_valgrind() -> bool {
+        env::var("ENABLE_VALGRIND").is_ok()
+    }
+
+    fn valgrind_test<F>(test_name: &str, test_body: F) -> Result<(), s2n_tls::error::Error>
+    where
+        F: FnOnce(&InstrumentationControl) -> Result<(), s2n_tls::error::Error>,
+    {
+        if !is_running_under_valgrind() {
+            let ctrl = InstrumentationControl;
+            test_body(&ctrl)
+        } else {
+            run_valgrind_test(test_name);
+            Ok(())
+        }
+    }
+
+    //test to create new config, set security policy, host_callback information, load/trust certs, and build config
+    #[test]
+    fn test_set_config() {
+        valgrind_test("test_set_config", |ctrl| {
+            ctrl.stop_instrumentation();
+            ctrl.start_instrumentation();
+            let keypair_rsa = CertKeyPair::default();
+            let _config =
+                set_config(&security::DEFAULT_TLS13, keypair_rsa).expect("Failed to build config");
+            Ok(())
+        })
+        .unwrap();
+    }
+
+    #[test]
+    fn test_rsa_handshake() {
+        valgrind_test("test_rsa_handshake", |ctrl| {
+            ctrl.stop_instrumentation();
+            // Example usage with RSA keypair (default)
+            let keypair_rsa = CertKeyPair::default();
+            let config = set_config(&security::DEFAULT_TLS13, keypair_rsa)?;
+            // Create a pair (client + server) using that config, start handshake measurement
+            let mut pair = TestPair::from_config(&config);
+            // Assert a successful handshake
+            ctrl.start_instrumentation();
+            assert!(pair.handshake().is_ok());
+            ctrl.stop_instrumentation();
+            Ok(())
+        })
+        .unwrap();
+    }
+    // function to run specified test using valgrind
+    fn run_valgrind_test(test_name: &str) {
+        let exe_path = std::env::args().next().unwrap();
+        let _dir_path = create_dir(Path::new("target/cg_artifacts"));
```

Comment:
```suggestion
        create_dir(Path::new("target/cg_artifacts")).unwrap();
```
I'm guessing that we probably don't want to totally ignore the return value? Since this is test code, it's totally fine to just panic on a weird/unexpected failure.